bisection.go

```diff
@@ -4,25 +4,38 @@
 
 package optimize
 
-import "math"
+import (
+	"math"
+
+	"github.com/gonum/floats"
+)
 
 // Bisection is a Linesearcher that uses a bisection to find a point that
 // satisfies the strong Wolfe conditions with the given gradient constant and
 // function constant of zero. If GradConst is zero, it will be set to a reasonable
 // value. Bisection will panic if GradConst is not between zero and one.
+// EqTol is used to smooth out floating point noise when testing the Wolfe
+// conditions. The Wolfe condition will be considered met if the function value
+// is less than or within a tolerance of the starting value of the line
+// search. If EqTol is zero, it will be defaulted to 1e-10, and if it is negative
+// a strict less than or equal to comparison will be used. A larger value of eqTol
+// will be more robust to numeric fluctuations but will miss real purturbations
+// in the objective function while a smaller value of eqTol will be more sensative
```

Comment:
If you keep this, then `sensative -> sensitive`.

When you edit this, could you move the whole EqTo to the field in the struct? And while you are at it, do the same for GradConst?